agent/stats/engine_integ_test.go

```diff
@@ -313,12 +374,16 @@ func TestStatsEngineWithDockerTaskEngine(t *testing.T) {
 			Container:  containers[0],
 		},
 		&testTask)
-	statsEngine := NewDockerStatsEngine(&cfg, dockerClient)
+
+	// Create a new docker stats engine
+	dockerStatsEngine = nil
```

Comment:
there's an opportunity here to rid us of some tech debt. Please add a todo in stats engine to remove this singleton object.